### PR TITLE
Switched to tsup to package for npm

### PR DESCRIPTION
**Pull request by JakeGinnivan**
Should fix an issue where bundled code is trying to use an export from tslib which doesn't exist

### CODE REVIEW

**Review by cwoodley — APPROVED** [review]

File: packages/json-react-layouts-data-loader/package.json [hunk 1]
```diff
@@ -5,9 +5,11 @@
     "author": "Seven West Media WA",
     "license": "MIT",
     "sideEffects": false,
-    "main": "dist/cjs/index.js",
-    "module": "dist/esm/index.js",
-    "devDependencies": {},
+    "main": "out-tsc/index.js",
+    "publishConfig": {
+        "main": "dist/cjs/index.js",
```
Comment (JakeGinnivan) [inline packages/json-react-layouts-data-loader/package.json]: dist/index.js

**Review by PatrickMilroy — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -38,7 +39,7 @@
         "react-ssr-data-loader": "^2.0.0-next.4",
         "ts-jest": "^27.0.4",
         "tsd": "^0.17.0",
-        "tslib": "^2.3.0",
+        "tsup": "^4.12.5",
```
Comment (unique3) [inline package.json:42]: 'CSS support is experimental.' 🤔

File: package.json [hunk 3]
```diff
@@ -6,12 +6,13 @@
     ],
     "scripts": {
         "build": "yarn tsc -b",
+        "package": "yarn tsup packages/json-react-layouts/src/index.ts --external debug --external react --target node14 --format cjs,esm --legacy-output -d packages/json-react-layouts/dist && yarn tsc -p packages/json-react-layouts --declaration --outDir packages/json-react-layouts/dist --emitDeclarationOnly && yarn tsup packages/json-react-layouts-data-loader/src/index.tsx --external debug --external react --target node14 --format cjs,esm --legacy-output -d packages/json-react-layouts-data-loader/dist && yarn tsc -p packages/json-react-layouts-data-loader --declaration --outDir packages/json-react-layouts-data-loader/dist --emitDeclarationOnly",
```
Comment (bennettp123) [inline package.json]: Is it possible to use `tsup --dts` to generate typedefs at build time? As per https://tsup.egoist.sh/#what-about-type-checking

If so, are there any advantages of `tsc -d --emitDeclarationOnly` over `tsup --dts`?

File: package.json (hunk 2 above)
Comment (bennettp123) [reply package.json:42]: How badly do we need css anyway 😛

**Review by unique3 — APPROVED** [review]

File: package.json (hunk 3 above)
Comment (bennettp123) [inline package.json]: Do we want to enable `--sourcemap`? What about `--minify`?

**Review by bennettp123 — APPROVED** [review]

File: package.json (hunk 2 above)
Comment (bennettp123) [reply package.json:42]: How badly do we need css anyway 😛

File: package.json (hunk 3 above)
Comment (JakeGinnivan) [reply package.json]: I can't get it to build using this option

Comment (JakeGinnivan) [reply package.json]: Will let the end bundler minify if they want, ill do sourcemaps

Comment (JakeGinnivan) [reply package.json]: I had issues when doing our nx-generator and this is the approach I used in the end there too

File: package.json (hunk 2 above)
Comment (JakeGinnivan) [reply package.json:42]: No css in sight :)

**Review by bennettp123 — APPROVED** [review]

**Pull request merged**